File: 102.二叉树的层序遍历.py
#
# @lc app=leetcode.cn id=102 lang=python3
#
# [102] 二叉树的层序遍历
#

# @lc code=start
# Definition for a binary tree node.


from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        result = []
        queue = [root]
        while queue:
            child = []
            nodes = []
            for node in queue:
                child.append(node.val)
                if node.left:
                    nodes.append(node.left)
                if node.right:
                    nodes.append(node.right)

            result.append(child)
            queue = nodes
        return result
# @lc code=end

File: 102.二叉树的层序遍历.js
/*
 * @lc app=leetcode.cn id=102 lang=javascript
 *
 * [102] 二叉树的层序遍历
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  if (!root) {
    return []
  }
  let queue = [root]
  let result = []
  while (queue.length) {
    let nodes = []
    let values = []
    for (const node of queue) {
      values.push(node.val)
      if (node.left) {
        nodes.push(node.left)
      }
      if (node.right) {
        nodes.push(node.right)
      }
    }
    result.push(values)
    queue = nodes
  }
  return result
};
// @lc code=end

